src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -61,29 +64,51 @@
 
     public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_MULTIPLE_PERSON = "More than 1 person exists with that name. Please look at the"
+            + "list below and enter the index of the client you wish to edit \n"
+            + "Example: 1, 2, 3 ...";
     public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the hustle book.";
+    public static final String MESSAGE_INVALID_INDEX = "This index does not exist!";
 
     private final Name targetName;
+    private final String targetNameStr;
     private final EditPersonDescriptor editPersonDescriptor;
+    private int index = 0;
 
     /**
      * @param name name of the person in the filtered person list to edit
      * @param editPersonDescriptor details to edit the person with
      */
-    public EditCommand(Name name, EditPersonDescriptor editPersonDescriptor) {
+    public EditCommand(String name, EditPersonDescriptor editPersonDescriptor) {
         requireNonNull(name);
         requireNonNull(editPersonDescriptor);
 
-        this.targetName = name;
+        this.targetNameStr = name;
+        this.targetName = new Name(name);
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
     }
```

Comment:
Maybe you can try this: 
```java 
    public EditCommand(Name name, EditPersonDescriptor editPersonDescriptor) {
        requireNonNull(name);
        requireNonNull(editPersonDescriptor);

        this.targetName = name;
        this.targetNameStr = name.fullName;
        this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
    }
```

This is to standardize the methods so any changes in the name in the future, we just modify the `Name` class instead of searching for all methods that use `String` instead of `Name`.